model-engine/model_engine_server/inference/forwarding/celery_forwarder.py

```diff
@@ -112,13 +118,17 @@ def after_return(
     # See documentation for options:
     # https://docs.celeryproject.org/en/stable/userguide/tasks.html#list-of-options
     @app.task(base=ErrorHandlingTask, name=LIRA_CELERY_TASK_NAME, track_started=True)
-    def exec_func(payload, *ignored_args, **ignored_kwargs):
+    def exec_func(payload, arrival_timestamp, *ignored_args, **ignored_kwargs):
         if len(ignored_args) > 0:
             logger.warning(f"Ignoring {len(ignored_args)} positional arguments: {ignored_args=}")
         if len(ignored_kwargs) > 0:
             logger.warning(f"Ignoring {len(ignored_kwargs)} keyword arguments: {ignored_kwargs=}")
         try:
-            return forwarder(payload)
+            result = forwarder(payload)
+            request_duration = datetime.now() - arrival_timestamp
+            if request_duration > timedelta(hours=12):  # longer than default visibility timeout
+                monitoring_metrics_gateway.emit_async_task_stuck_metric(queue_name)
```

Comment:
oh wait would we also want to emit "tasks not stuck" metrics? so we have some ratio